src/main/java/com/puppycrawl/tools/checkstyle/api/FullIdent.java

```diff
@@ -92,11 +91,16 @@ private static void extractFullIdent(FullIdent full, DetailAST ast) {
                 extractFullIdent(full, firstChild);
             }
             else if (typeOfAst == TokenTypes.DOT) {
-                final DetailAST firstChild = ast.getFirstChild();
-                extractFullIdent(full, firstChild);
-                full.append(".");
-                extractFullIdent(full, firstChild.getNextSibling());
-                appendBrackets(full, ast);
+                extractFullIdentFromDotAst(full, ast);
+            }
+            else if (typeOfAst == TokenTypes.METHOD_CALL) {
```

Comment:
We must generate regression reports for the following checks if there is an update to this method:

```
➜  checkstyle git:(master) grep -IR "createFullIdent" | grep -v "Test" | grep java | cut -d ":" -f 1 | uniq
src/main/java/com/puppycrawl/tools/checkstyle/api/FullIdent.java
src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/PackageNameCheck.java
src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/AbstractClassCouplingCheck.java
src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocMethodCheck.java
src/main/java/com/puppycrawl/tools/checkstyle/checks/design/FinalClassCheck.java
src/main/java/com/puppycrawl/tools/checkstyle/checks/design/VisibilityModifierCheck.java
src/main/java/com/puppycrawl/tools/checkstyle/checks/UncommentedMainCheck.java
src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalCatchCheck.java
src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalInstantiationCheck.java
src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EqualsHashCodeCheck.java
src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/PackageDeclarationCheck.java
src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnusedLocalVariableCheck.java
src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalThrowsCheck.java
src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/VariableDeclarationUsageDistanceCheck.java
src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/CovariantEqualsCheck.java
src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalTypeCheck.java
src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/CustomImportOrderCheck.java
src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/UnusedImportsCheck.java
src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/IllegalImportCheck.java
src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/AvoidStaticImportCheck.java
src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java
src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/RedundantImportCheck.java
src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportControlCheck.java
src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/AvoidStarImportCheck.java
src/main/java/com/puppycrawl/tools/checkstyle/utils/AnnotationUtil.java


```

Reply:
Please explain update of this method. Are we just passing in entire chained method call AST in one go?

Reply:
https://vyom-yadav.github.io/DiffReport/diffFullIdentChainedMethodCall/index.html
> Please explain update of this method.

FullIdent doesn't support typecast, chained method calls, and excess parentheses.

> Are we just passing in the entire chained method call AST in one go?

We pass the topmost method call (according to logic), get the name, and check it against the regex.